int32 floor
int32 door
bool in_door

